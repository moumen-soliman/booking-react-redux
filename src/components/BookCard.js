import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class BookCard extends Component {
    render() {
          const {id, title, image, description, author} = this.props
          return (
            <Link className="card-link" to={`/book/${id}`}> 
                <div className="card">
                    <div className="media">
                        <img className="mr-3" src={image} alt={title} width="200px"/>
                        <div className="media-body">
                            <h4 className="mt-0">{title}</h4> <span className="editmode-button" style={{display: this.props.editModeProp == 'false' ? 'inline-block' : 'none'}}>edit</span>
                            <p>{description}</p>
                        </div>
                    </div>
                    <hr />
                </div>
            </Link>
        )
    }
}

export default BookCard;