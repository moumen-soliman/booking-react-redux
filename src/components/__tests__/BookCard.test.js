import React from 'react';
import * as renderer from 'react-test-renderer';
import { MemoryRouter} from 'react-router';
import BookCard from '../BookCard';

describe('Book Card', () => {
    test('should render link output', () => {
        const component = renderer.create(
            <MemoryRouter> 
                <BookCard />
            </MemoryRouter>
        ).toJSON();

        expect(component).toMatchSnapshot();
    });
});

