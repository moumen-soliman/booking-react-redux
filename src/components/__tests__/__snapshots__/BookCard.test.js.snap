// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Card should render link output 1`] = `
<a
  className="card-link"
  href="/book/undefined"
  onClick={[Function]}
>
  <div
    className="card"
  >
    <div
      className="media"
    >
      <img
        className="mr-3"
        width="200px"
      />
      <div
        className="media-body"
      >
        <h4
          className="mt-0"
        />
         
        <span
          className="editmode-button"
          style={
            Object {
              "display": "none",
            }
          }
        >
          edit
        </span>
        <p />
      </div>
    </div>
    <hr />
  </div>
</a>
`;
